spark/graphar/src/main/java/org/apache/graphar/info/EdgeInfo.java

```diff
@@ -196,98 +205,73 @@ public PropertyGroup getPropertyGroup(String property) {
-    //     * @return A Result object containing the path prefix, or a Status object
-    //     * indicating an error.
-    //     */
-    //    public String getPropertyGroupPathPrefix(
-    //            PropertyGroup propertyGroup,
-    //            AdjListType adjListType) {
-    //
-    //    }
-
-    DataType getPropertyType(String propertyName) {
+    public String getPropertyGroupPrefix(PropertyGroup propertyGroup) {
+        checkPropertyGroupExist(propertyGroup);
+        return getPrefix() + "/" + propertyGroup.getPrefix();
+    }
+
+    public String getPropertyGroupChunkPath(PropertyGroup propertyGroup, long chunkIndex) {
+        // PropertyGroup will be checked in getPropertyGroupPrefix
+        return getPropertyGroupPrefix(propertyGroup) + "/chunk" + chunkIndex;
+    }
+
+    public String getAdjacentListPrefix(AdjListType adjListType) {
+        return getPrefix() + "/" + getAdjacentList(adjListType).getPrefix() + "/adj_list";
+    }
+
+    public String getAdjacentListChunkPath(AdjListType adjListType, long vertexChunkIndex) {
+        return getAdjacentListPrefix(adjListType) + "/chunk" + vertexChunkIndex;
+    }
+
+    public String getOffsetPrefix(AdjListType adjListType) {
+        return getAdjacentListPrefix(adjListType) + "/offset";
+    }
+
+    public String getOffsetChunkPath(AdjListType adjListType, long vertexChunkIndex) {
+        return getOffsetPrefix(adjListType) + "/chunk" + vertexChunkIndex;
+    }
+
+    public String getVerticesNumFilePath(AdjListType adjListType) {
+        return getAdjacentListPrefix(adjListType) + "/vertex_count";
+    }
+
+    public String getEdgesNumFilePath(AdjListType adjListType, long vertexChunkIndex) {
+        return getAdjacentListPrefix(adjListType) + "/edge_count" + vertexChunkIndex;
+    }
+
+    public DataType getPropertyType(String propertyName) {
         return propertyGroups.getPropertyType(propertyName);
     }
 
-    boolean isPrimaryKey(String propertyName) {
+    public boolean isPrimaryKey(String propertyName) {
         return propertyGroups.isPrimaryKey(propertyName);
     }
 
-    boolean isNullableKey(String propertyName) {
+    public boolean isNullableKey(String propertyName) {
         return propertyGroups.isNullableKey(propertyName);
     }
 
-    // TODO(@Thespica): Implement save and dump methods
-    //    /**
-    //     * Saves the edge info to a YAML file.
-    //     *
-    //     * @param fileName The name of the file to save to.
-    //     * @return A Status object indicating success or failure.
-    //     */
-    //    void save(String fileName) {
-    //
-    //    }
-    //
-    //    /**
-    //     * Returns the edge info as a YAML formatted string.
-    //     *
-    //     * @return A Result object containing the YAML string, or a Status object
-    //     * indicating an error.
-    //     */
-    //    public String dump() {
-    //
-    //    }
+    public void save(String filePath, Configuration conf) throws IOException {
```

Comment:
Ditto, the Configuration should be an optional parameter with default is null

Reply:
fixed